File: ExampleApp/app/src/main/java/com/motiv/example/GoApiApi.java
package com.motiv.example;

import retrofit2.*;
import retrofit2.http.*;

public interface GoApiApi {

    @GET("users/{userId}")
    Call<UserResponse> getUser(@Path("userId") java.lang.String userId);

    @GET("users")
    Call<UsersResponse> getUsersList();

    @GET("posts")
    Call<PostsListResponse> getPostsList();

    @GET("photos")
    Call<PhotosListResponse> getPhotos();
}

File: ExampleApp/app/src/main/java/com/motiv/example/GoApiApi.kt
 
package com.motiv.example
import retrofit2.*
import retrofit2.http.*

public interface GoApiApi {

    @GET("users/{userId}")
    fun getUser(
        @Path("userId")
        userId: String
    ): Call<UserResponse>@GET("users")
    fun getUsersList(): Call<UsersResponse>@GET("posts")
    fun getPostsList(): Call<PostsListResponse>@GET("photos")
    fun getPhotos(): Call<PhotosListResponse>
}
